client/my-sites/pages/page/index.js

```diff
@@ -57,56 +67,38 @@ const Page = React.createClass( {
 		previousStatus: PropTypes.string,
 		showMoreOptions: PropTypes.bool.isRequired,
 		showPageActions: PropTypes.bool.isRequired,
-	},
-
-	analyticsEvents: {
-		moreOptions: function() {
-			recordEvent( 'Clicked More Options Menu' );
-		},
-		pageTitle: function() {
-			recordEvent( 'Clicked Page Title' );
-		},
-		editPage: function() {
-			recordEvent( 'Clicked Edit Page' );
-		},
-		viewPage: function() {
-			recordEvent( 'Clicked View Page' );
-		},
-		statsPage: function() {
-			recordEvent( 'Clicked Stats Page' );
-		}
-	},
+	}
 
-	togglePageActions: function() {
+	togglePageActions = () => {
 		this.props.togglePageActions();
 		// TODO check previous impl for race conditions
 		if ( this.props.showPageActions ) {
-			this.analyticsEvents.moreOptions();
+			this.props.recordMoreOptions();
 		}
-	},
+	}
 
 	// Construct a link to the Site the page belongs too
-	getSiteDomain: function() {
+	getSiteDomain() {
 		return ( this.props.site && this.props.site.domain ) || '...';
-	},
+	}
 
-	viewPage: function( event ) {
+	viewPage = ( event ) => {
 		event.preventDefault();
 		const { isPreviewable, previewURL } = this.props;
 
 		if ( this.props.page.status && this.props.page.status === 'publish' ) {
-			this.analyticsEvents.viewPage();
+			this.props.recordViewPage();
```

Comment:
Actually, now I know why I didn't do this:

- there's a certain quality to calling bound action creators by their _full name_, so to speak: the nature of `this.props.recordViewPage()` is very evident — you know it's a prop, but because it's a function call (and not named `onViewPage`) you can tell that it's a bound action creator

- that alone is not a great point, *but* in most cases you'd have variable shadowing that ESLint doesn't like; for instance, two lines below,

```js
this.props.setPreviewUrl( previewURL );
this.props.setLayoutFocus( 'preview' );
```

if I were to extract them from `this.props`, ESLint would give me

```
'setPreviewUrl' is already declared in the upper scope. (no-shadow)
```

`recordViewPage` is an exception because it was created in `mapDispatch`, but out of consistency it seems best to preserve the "full path".